airflow/auth/managers/base_auth_manager.py

```diff
@@ -63,6 +68,69 @@ def get_user_id(self) -> str:
     def is_logged_in(self) -> bool:
         """Return whether the user is logged in."""
 
+    @abstractmethod
+    def is_authorized(
+        self,
+        action: ResourceAction,
+        resource_type: str,
+        resource_details: ResourceDetails | None = None,
+        user: BaseUser | None = None,
+    ) -> bool:
+        """
+        Return whether the user is authorized to perform a given action.
+
+        .. code-block:: python
+
+            # Check whether the logged-in user has permission to read the DAG "my_dag_id"
+            get_auth_manager().is_authorized(
+                Action.GET,
+                Resource.DAG,
+                ResourceDetails(
+                    id="my_dag_id",
+                ),
+            )
+
+        :param action: the action to perform
+        :param resource_type: the type of resource the user attempts to perform the action on
+        :param resource_details: optional details about the resource itself
+        :param user: the user to perform the action on. If not provided (or None), it uses the current user
+        """
+
+    def is_all_authorized(
+        self,
+        actions: Sequence[AuthorizedAction],
+    ) -> bool:
+        """
+        Wrapper around `is_authorized` to check whether the user is authorized to perform several actions.
+
+        :param actions: the list of actions to check. Each item represents the list of parameters of
+            `is_authorized`
+        """
+        return all(
+            self.is_authorized(
+                action=action.action,
+                resource_type=action.resource_type,
+                resource_details=action.resource_details,
+                user=action.user,
+            )
+            for action in actions
+        )
+
+    def _get_root_dag_id(self, dag_id: str) -> str:
+        """
+        Return the root DAG id in case of sub DAG, return the DAG id otherwise.
+
+        :param dag_id: the DAG id
+        """
+        if "." in dag_id:
```

Comment:
Actually.. I am thinking this might be a good point to stop supporting subdags. SubDAGs are deprecated and we are supposed to discourage them. I would say that making AIP-56 not supporting SubDAGs is a way better than supporting them. 

Why don't we say that SubDAGs are not supported with AIP-56, failing them outright and raising exception where we ask the user to convert to TaskGroup.

I think one of the reasons fo AIP-56 is to implement everything in a cleaner way than all the options today.

I am not sure how that would play with FAB auth, but I'd say "stop worrying about adding support in new features for deprecated features".

Reply:
Sounds good to me then :)

Reply:
It **might** become an implementation detail of a FAB Auth Manager. I would have completely no problem wth it.

Reply:
I moved it to FAB auth manager to keep it backward compatible. As you said, it is now an implementation detail of FAB auth manager